tests/python_package_test/test_engine.py

```diff
@@ -1545,6 +1545,40 @@ def test_refit():
     assert err_pred > new_err_pred
 
 
+def test_refit_dataset_params():
+    # check refit accepts dataset_params
+    X, y = load_breast_cancer(return_X_y=True)
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=42)
```

Comment:
Do we really need this split? I think it would be enough to check that the predictions over the training set are different between the original and the refitted booster.

Reply:
Both work, we just want to test that predictions values or metric changes when calling `refit`, it can be in `train` or `test` set.

I think we can add an additional test to proof the change in the `train` set, what are your thoughts @jmoralez ?

Reply:
> I think we can add an additional test to proof the change in the train set

You don't need to add a second test. I agree with @jmoralez ... calling `.predict()` on any data will be enough for this unit test. Please remove the `train_test_split()` and just create predictions on `X`.

Reply:
got it!